src/room/InCallView.tsx

```diff
@@ -85,29 +85,43 @@ import {
   ECAddonConnectionState,
   ECConnectionState,
 } from "../livekit/useECConnectionState";
+import { useOpenIDSFU } from "../livekit/openIDSFU";
 
 const canScreenshare = "getDisplayMedia" in (navigator.mediaDevices ?? {});
 const isSafari = /^((?!chrome|android).)*safari/i.test(navigator.userAgent);
 
 // How long we wait after a focus switch before showing the real participant list again
 const POST_FOCUS_PARTICIPANT_UPDATE_DELAY_MS = 3000;
 
-export interface ActiveCallProps extends InCallViewProps {
+export interface ActiveCallProps
+  extends Omit<InCallViewProps, "livekitRoom" | "connState"> {
   e2eeConfig: E2EEConfig;
 }
 
 export const ActiveCall: FC<ActiveCallProps> = (props) => {
-  if (!props.livekitRoom) {
-    return null;
-  }
+  const sfuConfig = useOpenIDSFU(props.client, props.rtcSession);
+  const { livekitRoom, connState } = useLiveKit(
+    props.rtcSession,
+    props.muteStates,
+    sfuConfig,
+    props.e2eeConfig,
+  );
+
+  const cleanup = useCallback(() => {
+    if (connState === ConnectionState.Connected) {
+      livekitRoom?.disconnect();
+    }
+  }, [livekitRoom, connState]);
+
+  useEffect(() => {
+    return cleanup;
+  }, [cleanup, livekitRoom]);
```

Comment:
Note that:
1. This clean-up function may be called multiple times.
2. The connection state could also be `Reconnecting` or `Connecting` in which case the `disconnect()` will not be called (meaning that we will connect to the room despite unmounting the `ActiveCall` leading to an unexpected bug for this edge-case.

I think the cleanest way to fix the problem is to do a clean effect that just disconnects the room on unmount without any further actions and checks (I don't think we need `connState` here), similarly to how LiveKit does it. I checked the LiveKit Meet example and how they do it. They use the `LiveKitRoom` component that relies on `LkRoomContext`. Here is how the clean-up is done: [link](https://github.com/livekit/components-js/blob/c5305c35cca3b8d08fc9f9652ef48acaa2f3db55/docs/storybook/.storybook/lk-decorators/RoomContext.tsx#L19C1-L19C1) which is just a typical React idiom for these sort of cases. Are there any strong reasons why we don't want to do it this way? 🙂

Reply:
I had this initially. But was seeing it beeing called in the beginning. But that was because i let copilot + tooling add the dependencies...

Reply:
I now use exactly what you propose.